### PR TITLE
parser: Charset name can be string literal

### PR DESCRIPTION
**Pull request by shenli**
Fix bug in running phpMyAdmin

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1009,6 +1009,16 @@ CharsetName:
 			return 1
 		}
 	}
+|	stringLit
```
Comment (ngaut) [inline parser/parser.y]: Rename stringLit to stringLiteral may be better.

File: parser/parser.y [hunk 2]
```diff
@@ -1009,6 +1009,16 @@ CharsetName:
...
+	{
+		c := strings.ToLower($1.(string))
+			if charset.ValidCharsetAndCollation(c, "") {
```
Comment (qiuyesuifeng) [inline parser/parser.y]: Alignment.

Comment (shenli) [inline parser/parser.y]: Ok

File: parser/parser.y (hunk 1 above)
Comment (shenli) [inline parser/parser.y]: There are also intLit/floatLit. I will clean them in the next pr.

**Comment by shenli** [comment]
PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**